app/code/Magento/Sales/Model/Order/OrderHistory.php

```diff
@@ -0,0 +1,175 @@
+            }
+        }
+
+        return $history;
+    }
+
+    /**
+     * Get shipment history data for specific order
+     *
+     * @param \Magento\Sales\Model\Order $order
+     * @return array
+     */
+    public function getShipmentHistory($order)
+    {
+        $history = [];
+        foreach ($order->getShipmentsCollection() as $_shipment) {
+            $history[] = $this->_prepareHistoryItem(
+                __('Shipment #%1 created', $_shipment->getIncrementId()),
+                $_shipment->getEmailSent(),
+                $_shipment->getCreatedAt()
+            );
+
+            foreach ($_shipment->getCommentsCollection() as $_comment) {
+                $history[] = $this->_prepareHistoryItem(
+                    __('Shipment #%1 comment added', $_shipment->getIncrementId()),
+                    $_comment->getIsCustomerNotified(),
+                    $_comment->getCreatedAt(),
+                    $_comment->getComment()
+                );
+            }
+        }
+
+        return $history;
+    }
+
+    /**
+     * Get invoice history data for specific order
+     *
+     * @param \Magento\Sales\Model\Order $order
+     * @return array
+     */
+    public function getInvoiceHistory($order)
+    {
+        $history = [];
+        foreach ($order->getInvoiceCollection() as $_invoice) {
+            $history[] = $this->_prepareHistoryItem(
+                __('Invoice #%1 created', $_invoice->getIncrementId()),
+                $_invoice->getEmailSent(),
+                $_invoice->getCreatedAt()
+            );
+
+            foreach ($_invoice->getCommentsCollection() as $_comment) {
+                $history[] = $this->_prepareHistoryItem(
+                    __('Invoice #%1 comment added', $_invoice->getIncrementId()),
+                    $_comment->getIsCustomerNotified(),
+                    $_comment->getCreatedAt(),
+                    $_comment->getComment()
+                );
+            }
+        }
+
+        return $history;
+    }
+
+    /**
+     * Get tracking history data for specific order
+     *
+     * @param \Magento\Sales\Model\Order $order
+     * @return array
+     */
+    public function getTracksHistory($order)
+    {
+        $history = [];
+        foreach ($order->getTracksCollection() as $_track) {
+            $history[] = $this->_prepareHistoryItem(
+                __('Tracking number %1 for %2 assigned', $_track->getNumber(), $_track->getTitle()),
+                false,
+                $_track->getCreatedAt()
+            );
+        }
+
+        return $history;
+    }
+
+    /**
+     * Compose and get order full history.
+     * Consists of the status history comments as well as of invoices, shipments and creditmemos creations
+     *
+     * @param \Magento\Sales\Model\Order $order
+     * @return array
+     */
+    public function getFullHistory($order)
+    {
+        $history = [];
+        $history = array_merge($history, $this->getAllStatusHistory($order));
```

Comment:
According to [array_merge documentation](http://php.net/manual/en/function.array-merge.php) you can pass to array_merge function one or more arrays, so, you can simplify this expression: `$history = array_merge($this->getInvoiceHistory(), $this->getShipmentHistory(), etc);`